docs/en/Logging-Conventions.md

```diff
@@ -0,0 +1,101 @@
+
+Debug Level Logging
+
+* Debug level logs include detailed information for various aspects of the Alluxio system.
+* Control flow logging (Alluxio system enter and exit calls) is done in debug level logs.
+
+Trace Level Logging
+
+* Trace level logs are not used in Alluxio.
+
+## Configuration
+
+Alluxio's logging behavior can be fully configured through the `log4j.properties` file found in the
+`conf` folder.
+
+By default, Alluxio logs to files in the `logs` directory which can be modified by setting the
+`alluxio.logs.dir` configuration value. Each Alluxio process (Master, Worker, Clients, FUSE, Proxy)
+logs to a different file.
+
+### Remote Logging
+
+By default, Alluxio processes log to local files. In certain environments, it is more reliable to
+log to a central machine. Alluxio supports using Log4J's
+[SocketAppender](https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/net/SocketAppender.html)
+to achieve this.
+
+To set up remote logging, first determine the host which you plan to aggregate the logs at. On that
+host, download the [Log4J jar](https://mvnrepository.com/artifact/log4j/log4j/1.2.17) which contains
+a server implementation.
+
+The server requires a `log4j.properties` file, use the file provided in Alluxio's `conf` folder. In
+the file, replace all instances of `{alluxio.*}` variables with complete values, for example,
+instead of `{alluxio.logs.dir}`, use `/opt/alluxio/logs`.
+
+Then start the server with the following command
+
+```bash
+java -cp <PATH_TO_LOG4J_JAR> org.apache.log4j.net.SimpleSocketServer <PORT> <PATH_TO_LOG4J_PROPERTIES>
+```
+
+The server is now ready to log any incoming SocketAppender traffic. You can validate this by looking
+in the logs directory, the first two lines in the log files should indicate the server was started
+succcessfully.
+
+Since this will be a long running process, it is worth considering using a separate service to
+ensure the server is up. If the server is unavailable, Alluxio processes will not be affected but
+logs will be lost.
+
+On the Alluxio process side, update `conf/log4j.properties`, for example for `MASTER_LOGGER` replace
+the existing properties with:
+
+```
+log4j.appender.MASTER_LOGGER=org.apache.log4j.net.SocketAppender
+log4j.appender.MASTER_LOGGER.Port=<PORT>
+log4j.appender.MASTER_LOGGER.RemoteHost=<HOSTNAME_OF_LOG_SERVER>
+log4j.appender.MASTER_LOGGER.ReconnectionDelay=<MILlIS_TO_WAIT_BEFORE_RECONNECTION_ATTEMPT>
+log4j.appender.MASTER_LOGGER.layout=org.apache.log4j.PatternLayout
+log4j.appender.MASTER_LOGGER.layout.ConversionPattern=%d{ISO8601} %-5p %c{1} - %m%n
+```
+
+Then restart your Alluxio processes to pick up the latest logging configurations. Logs should now be
+redirected to the remote server instead of being logged to a local file.
+
+Note that logging to a remote server results in logs being aggregated instead of at a per machine
+level. In addition, it is often beneficial to log to both the local and remote machines.
```

Comment:
I added an example. I don't think we should aim to provide an example for every use case though, since there are many of those and its mostly related to setting up log4j and not Alluxio.